arviz/tests/test_data.py

```diff
@@ -640,25 +675,41 @@ def test_io_function(self, data, eight_schools_params):
         assert os.path.exists(filepath)
         assert os.path.getsize(filepath) > 0
         inference_data2 = from_netcdf(filepath)
-        assert hasattr(inference_data2, "posterior")
+        fails = check_multiple_attrs(test_dict, inference_data2)
+        assert not fails
         os.remove(filepath)
         assert not os.path.exists(filepath)
 
-    def test_io_method(self, data, eight_schools_params):
+    @pytest.mark.parametrize("groups_arg", [False, True])
+    def test_io_method(self, data, eight_schools_params, groups_arg):
         inference_data = self.get_inference_data(  # pylint: disable=W0612
             data, eight_schools_params
         )
-        assert hasattr(inference_data, "posterior")
+        if groups_arg:

```

Comment:
Same thing here this test is getting pretty big actually. is there a way to separate the new assert into a new test?

Reply:
Yes, I tried to put them toghether because they are both testing the same function with one different argument. They can be divided without much trouble between `test_io_method` and `test_io_method_groups`

Reply:
How does it look now? I tried to minimize the parts where using a `groups` list and `None` diverged, because separating them was repeating a lot of equal code.